src/gwt/src/org/rstudio/studio/client/workbench/views/source/editors/EditingTargetCodeExecution.java

```diff
@@ -229,7 +231,28 @@ public void sendSelectionToTerminal(boolean skipBlankLines)
          moveCursorAfterExecution(selectionRange, skipBlankLines);
       }
    }
-   
+
+   public void openNewTerminalHere()
+   {
+      if (target_ == null)
+         return;
+      if (StringUtil.isNullOrEmpty(target_.getPath()))
+         return;
+      FileSystemItem file = FileSystemItem.createFile(target_.getPath());
+      events_.fireEvent(new CreateNewTerminalEvent(file.getParentPathString()));
+   }
+
+   public void sendFilenameToTerminal()
+   {
+      if (target_ == null)
+         return;
+      String filename = target_.getPath();
+      if (StringUtil.isNullOrEmpty(filename))
+         filename = target_.getName().getValue();
+      if (!StringUtil.isNullOrEmpty(filename))
+         events_.fireEvent(new SendToTerminalEvent(filename, true /*focus terminal*/));
```

Comment:
It will be a no-op. This is pretty common, lots of places where we fire commands and drop them on the floor. Mainly because we don't consistently enable/disable commands, so easy to get into cases where you can trigger a command (e.g. via a keyboard shortcut) when it doesn't make sense. The defensive programming here is to just avoid JavaScript exceptions.

Reply:
Actually, I think I've already hidden this command in cases where it isn't applicable, but earlier in development I hadn't. So yes I could show a message, or remove the checks, but I'd rather err on the side of not throwing Javascript exceptions if there are still cases where a command is active when it shouldn't be (and generally the command makes no sense in those situation so user wouldn't have expected it to do anything, probably accidentally hit the shortcut, so a message would be confusing).

Reply:
Makes sense to me!